src/hotspot/share/cds/aotLinkedClassBulkLoader.cpp

```diff
@@ -210,3 +231,29 @@ void AOTLinkedClassBulkLoader::initiate_loading(JavaThread* current, const char*
     }
   }
 }
+
+void AOTLinkedClassBulkLoader::finish_loading_javabase_classes(TRAPS) {
+  init_required_classes_for_loader(Handle(), AOTLinkedClassTable::for_static_archive()->boot(), CHECK);
+}
+
+// Some AOT-linked classes for <class_loader> must be initialized early. This includes
+// - classes that were AOT-initialized by AOTClassInitializer
+// - the classes of all objects that are reachable from the archived mirrors of
+//   the AOT-linked classes for <class_loader>.
```

Comment:
It seems this function covers the first two categories, but not the AOT-linked classes. Is that correct?

Reply:
Correct. Not all aot-linked classes are initialized by this function. Most them are initialized on first use.